src/Features/Core/Portable/Completion/Providers/AbstractAwaitCompletionProvider.cs

```diff
@@ -92,18 +140,33 @@ private protected bool ShouldMakeContainerAsync(SyntaxToken token, SyntaxGenerat
             return declaration is not null && !generator.GetModifiers(declaration).IsAsync;
         }
 
-        private CompletionItem GetCompletionItem(SyntaxToken token, SyntaxGenerator generator, ISyntaxKindsService syntaxKinds, ISyntaxFactsService syntaxFacts)
+        private CompletionItem GetCompletionItem(SyntaxToken token, bool isAwaitKeywordContext, bool isDotAwaitContext, SyntaxGenerator generator, ISyntaxKindsService syntaxKinds, ISyntaxFactsService syntaxFacts)
         {
             var shouldMakeContainerAsync = ShouldMakeContainerAsync(token, generator);
             var text = syntaxFacts.GetText(syntaxKinds.AwaitKeyword);
+            var completionChangeEdit = GetCompletionChangeEdit(isAwaitKeywordContext, isDotAwaitContext, shouldMakeContainerAsync);
+
             return CommonCompletionItem.Create(
                 displayText: text,
                 displayTextSuffix: "",
                 rules: CompletionItemRules.Default,
                 Glyph.Keyword,
                 description: RecommendedKeyword.CreateDisplayParts(text, FeaturesResources.Asynchronously_waits_for_the_task_to_finish),
                 inlineDescription: shouldMakeContainerAsync ? FeaturesResources.Make_containing_scope_async : null,
-                isComplexTextEdit: shouldMakeContainerAsync);
+                isComplexTextEdit: shouldMakeContainerAsync || isDotAwaitContext,
+                properties: ImmutableDictionary.Create<string, string>().Add(nameof(CompletionChangeEdit), completionChangeEdit.ToString()));
+
+            static CompletionChangeEdit GetCompletionChangeEdit(bool isAwaitKeywordContext, bool isDotAwaitContext, bool shouldMakeContainerAsync)
+            {
+                var codeActionEdit = default(CompletionChangeEdit);
+                if (isAwaitKeywordContext)
+                    codeActionEdit |= CompletionChangeEdit.AddAwaitAtCursor;
+                if (isDotAwaitContext)
+                    codeActionEdit |= CompletionChangeEdit.AddAwaitBeforeDotExpression;
+                if (shouldMakeContainerAsync)
+                    codeActionEdit |= CompletionChangeEdit.MakeContainerAsync;
+                return codeActionEdit;
```

Comment:
i'm fine with this.  however, another (potentially simpler) way to do this woudl just to be to have three separate properties on the prop dictionary.  then we dont' need the enum, and we don't have to have special toString+parse behavior (Admittedly you get that for free with enums here, but it still seems a bit overkill to me at elast).